HUDOC.js

```diff
@@ -0,0 +1,502 @@
+- Handle more than judgments or decisions (summaries, translations, search results)
+- Handle reports
+- Handle resolutions and advisory opinions properly (esp. title)
+- Handle different types of documents via the API.
+- Use keywords for tags
+- Use references to create "related" entries if they exist
+- Put in short titles
+
+=> All relate to one thing: This translator is currently unable to access the API
+to query it for the taxonomies/thesauruses used for decision type ("typedescription"),
+decision-making body ("originatingbody"), keywords ("kpthesaurus"), ...
+Check compiled.js on the HUDOC website in order to learn how to query these aspects.
+Search for "// ###..//sites/echr/echr.js ###"" in compiled.js to find most API fields.
+*/
+
+function text(docOrElem, selector, index) { var elem = index ? docOrElem.querySelectorAll(selector).item(index) : docOrElem.querySelector(selector); return elem ? elem.textContent: null; }
+
+// Scrapes some metadata from the document
+// TODO: integrate function into scrape
+function scrapeMetaData(doc, detail) { // Only scrapes the header of the main page
+	switch (detail) {
+		case "appno": return text(doc, "span.column01");
+		case "typedescription": return text(doc, "span.column02");
+		case "originatingbody": return text(doc, "span.column03");
+		case "judgementdate": return text(doc, "span.column04");
+		default: return null;
+	}
+}
+
+// Gets the itemid of the document
+function getItemID(url) {
+	var urlRegex = /%22itemid%22:\[%22([\d-]*)%22]|\?i=([\d-]*)/i;
+	var id = url.match(urlRegex);
+	return id[1];
+}
+
+// Adds the type of judgment at the end of the name
+function getTypeBit(doc, url) { // TODO: Switch to 'url' once we use the API instead
+	var description = scrapeMetaData(doc, "typedescription");
+	var language = scrapeMetaData(doc, "typedescription");
+
+	//Simpler logic for French that just takes over the type (unless it's a judgment)
+	// TODO: Add comprehensive logic like for English
+	if (url.includes("hudoc.echr.coe.int/fre#")) {
+		if (description.includes("Arrêt")) {
+			return "";
+		}
+		else {
+			return " (" + description.split("(")[0].toLowerCase() + ")";
+		}
+	}
+
+	if (description.includes("Judgment")) {
+		if (description.includes("Just Satisfaction") && !description.includes("Merits")) {
+			return " (just satisfaction)";
+		}
+		if (description.includes("Objection") || description.includes("Jurisdiction")) {
+			return " (preliminary objections)";
+		}
+		// TODO: Distinguish " (friendly settlement)" and " (striking out)"
+		// 				Requires a queryMetaData function
+		if (description.includes("Struck out")) return " (striking out)";
+		if (description.includes("Interpretation"))	return " (interpretation)";
+		if (description.includes("Revision")) return " (revision)";
+
+		return "";
+	}
+
+	if (description.includes("Decision")) return " (dec.)";
+	if (description.includes("Communicated")) return " (communicated)"; // TODO: Check if correct
```

Comment:
What format do these need to follow? Were you able to figure out if this is correct?

Reply:
Seems to agree with most of the literature